redbot/cogs/mod/names.py

```diff
@@ -69,6 +69,96 @@ class ModInfo(MixinMeta):
             else:
                 await ctx.send(_("Done."))
 
+    def handle_custom(self, user):
+        a = [c for c in user.activities if c.type == discord.ActivityType.custom]
+        if not a:
+            return None, discord.ActivityType.custom
+        a = a[0]
+        c_status = None
+        if not a.name and not a.emoji:
+            return None, discord.ActivityType.custom
+        elif a.name and a.emoji:
+            c_status = _("Custom: {emoji} {name}").format(emoji=a.emoji, name=a.name)
+        elif a.emoji:
+            c_status = _("Custom: {emoji}").format(emoji=a.emoji)
+        elif a.name:
+            c_status = _("Custom: {name}").format(name=a.name)
+        return c_status, discord.ActivityType.custom
+
+    def handle_playing(self, user):
+        p_acts = [c for c in user.activities if c.type == discord.ActivityType.playing]
+        p_act = p_acts[0] if p_acts else None
+        act = (
+            _("Playing: {name}").format(name=p_act.name if p_act and p_act.name else None)
+            if p_act and hasattr(p_act, "name")
+            else p_act.name
+            if p_act and p_act.name
+            else None
+        )
+        return act, discord.ActivityType.playing
+
+    def handle_streaming(self, user):
+        s_acts = [c for c in user.activities if c.type == discord.ActivityType.streaming]
+        s_act = s_acts[0] if s_acts else None
+        act = (
+            _("Streaming on {platform}: [{name}{sep}{game}]({url})").format(
+                platform=s_act.platform,
+                name=discord.utils.escape_markdown(s_act.name),
+                sep=" | " if s_act.game else "",
+                game=discord.utils.escape_markdown(s_act.game) if s_act.game else "",
+                url=s_act.url,
+            )
+            if s_act and s_act.name and hasattr(s_act, "game")
+            else discord.utils.escape_markdown(s_act.name)
+            if s_act and s_act.name
+            else None
+        )
```

Comment:
When activity has a type `streaming`, it can either have url and then it will have `discord.Streaming` type, or it might also not have the url and then it will be `discord.Activity`. For the first, you can get quite a lot of info about the activity and you make use of that, however we also need to handle the second case. I suggest handling both of those differently using `isinstance()`:
```suggestion
        if not s_acts:
            return None, discord.ActivityType.streaming
        s_act = s_acts[0]
        if isinstance(s_act, discord.Streaming):
            act = _("Streaming on {platform}: [{name}{sep}{game}]({url})").format(
                platform=s_act.platform,
                name=discord.utils.escape_markdown(s_act.name),
                sep=" | " if s_act.game else "",
                game=discord.utils.escape_markdown(s_act.game) if s_act.game else "",
                url=s_act.url,
            )
        else:
            act = _("Streaming: {name}").format(name=s_act.name)
```